lib/temporal/workflow/query_registry.rb

```diff
@@ -19,7 +19,10 @@ def handle(type, args = nil)
         handler = handlers[type]
 
         unless handler
-          raise Temporal::QueryFailed, "Workflow did not register a handler for #{type}"
+          # The end of the formatted error message (e.g., "KnownQueryTypes=[query-1, query-2, query-3]")
+          # is used by temporal-web to show a list of queries that can be run on the 'Query' tab of a
+          # workflow. If that part of the error message is changed, that functionality will break.
+          raise Temporal::QueryFailed, "Workflow did not register a handler for '#{type}'. KnownQueryTypes=[#{handlers.keys.join(", ")}]"
```

Comment:
Nice find - this makes much more sense to me now. I remember seeing the `KnownQueryTypes=...` format included in the Go SDK error message and didn't make the connection. 🤦

Reply:
It took me some time to wrap my brain around this as well. Looks like it goes way back to Cadence https://github.com/uber-go/cadence-client/pull/226